src/Common/AsynchronousMetrics.cpp

```diff
@@ -1049,6 +1043,14 @@ void AsynchronousMetrics::update(TimePoint update_time, bool force_update)
 
                 if (name.starts_with("cpu"))
                 {
+                    if (cgroup_cpu_metrics_present)
```

Comment:
Wouldn't it be simpler always to read `ProcStatValuesCPU` first (it doesn't depend on the number of CPU, so the check below is irrelevant)
```suggestion
    ProcStatValuesCPU current_values{};
    current_values.read(*proc_stat);
    
    /// Skip the CPU metrics if we already have them from cgroup
    if (cgroup_cpu_metrics_present)
        continue;
```